grpc-interface-support/src/main.py

```diff
@@ -25,12 +26,55 @@
 from velocitas_lib import (
     get_programming_language,
     get_project_cache_dir,
+    get_workspace_dir,
     obtain_local_file_path,
 )
 from velocitas_lib.functional_interface import get_interfaces_for_type
-from velocitas_lib.text_utils import create_truncated_string
 
 DEPENDENCY_TYPE_KEY = "grpc-interface"
+DOWNLOAD_PATH = os.path.join(get_project_cache_dir(), "downloads")
+
+
+def extract_zip(file_path: str, extract_to: str) -> str:
+    """Extract a zip file.
+
+    Args:
+        file_path (str): The file path to the zip.
+        extract_to (str): The file path to extract to.
+
+    Raises:
+        RuntimeError if the file_path is not a zip.
+
+    Returns:
+        str: The file path to the extracted top level folder.
+    """
+    if zipfile.is_zipfile(file_path):
+        with zipfile.ZipFile(file_path, "r") as zip_ref:
+            zip_ref.extractall(extract_to)
+
+        return extract_to
+    else:
+        raise RuntimeError(f"{file_path!r} is not a zip file!")
+
+
+def discover_service_proto_files_in_filetree(
```

Comment:
this is discovering all protos not only the ones containing a service, hence I suggest
```suggestion
def discover_proto_files_in_filetree(
```